### PR TITLE
feat(products.controller) :

### PR DESCRIPTION
**Pull request by ShumerPlema**
## Related task

- [2.1] https://github.com/ITA-Dnipro/Dp-202-shop/issues/14

## Changes

- Added getOneProductById method  which gives information about the product by the selected id, and also accepts a true or false argument to display deleted products. If true - removed products are shown. False - not shown;
- Added getAllProducts method that shows all NOT deleted products;
- Added getAllProducts method, which shows all products with all information except ("unit_id", "manufacture_id", "category_id", "user_id");
- Added idIsExist method to check if id exists. It also accepts true or false as an argument to display remote products.
- Changed error-handler.middleware due to incorrect error catching

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # dependencies
 ./Dp-202-shop/server/node_modules/
 ./Dp-202-shop/server/dist/
-./Dp-202-shop/node_modules/
```
Comment (kolesnicknick) [inline .gitignore:4]: return this back

File: Dp-202-shop/server/src/modules/products/product.service.ts [hunk 2]
```diff
@@ -1,13 +1,13 @@
-import { Manufacture } from '../../db/models/Manufacture.model';
-import { Category } from '../../db/models/Category.model';
-import { normalize, normalizeOne } from '../../common/helpers/dataNormalization';
-import { Unit } from '../../db/models/Unit.model';
-import { User } from '../../db/models/User.model';
-import { Op } from 'sequelize';
-import { ProductAttributes } from '../../db/models/Product.model';
-import { userService } from '../users/user.service';
-import { Product } from '../../db/models/Product.model';
-import { NotFoundData } from '../../common/errors/notFoundData';
```
Comment (kolesnicknick) [inline Dp-202-shop/server/src/modules/products/product.service.ts]: return proper formation back

**Review by kolesnicknick — CHANGES_REQUESTED** [review]
return formation back. commit only lines you have changed

**Review by valeriia-danylenko — APPROVED** [review]
Everything works and that's great

**Review by DianaBondarenko — APPROVED** [review]

**Pull request merged**